spectacles/lookml.py

```diff
@@ -322,14 +325,46 @@ def get_explore(self, model: str, name: str) -> Optional[Explore]:
             return model_object.get_explore(name)
 
     def get_results(
-        self, validator: str, fail_fast: Optional[bool] = None
+        self,
+        validator: str,
+        fail_fast: Optional[bool] = None,
+        filters: Optional[List[str]] = None,
     ) -> Dict[str, Any]:
         errors: List[Dict[str, Any]] = []
         successes: List[Dict[str, Any]] = []
         tested = []
 
         for model in self.models:
+            # Add model level content validation errors.
+            # We create an explore "tested" record for those that
+            # aren't in the LookML tree.
+            distinct_explores = set()
+            if filters is None and model.errors:
+                # Raise name error if there are errors on the model and filters
+                # isn't being passed. We need filters when running the content
+                # validator, which is the only time a model will have errors.
+                raise NameError(
+                    "The argument 'filters' is required in Content Validator."
+                )
+            elif filters is not None and model.errors:  # appease mypy
+                for error in [
+                    e
+                    for e in model.errors
+                    if is_selected(model.name, e.explore, filters)
+                ]:
```

Comment:
Nit: Generally not a fan of looping over a conditional list comprehension like this. I think it's more straightforward to write a plain `for` loop with a nested `if`, that way you only read one `for` instead of two.